pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -165,6 +165,16 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 		}
 	}
 
+	// Check if pods container runs the right image
```

Comment:
What's this check about?

Reply:
Suppose that we have a 9.4.0 running cluster and the user changes the image to 9.4.5 and deploys cr. The Reconcile policy needs to handle this in some way, but we need some use cases definition first: i.e. is this a cluster upgrade use case?

Basically I added this check as a todo flag atm.